library/data/recipe/src/main/java/org/quiltmc/qsl/recipe/mixin/BrewingStandBlockEntityMixin.java

```diff
@@ -0,0 +1,153 @@
+	@Redirect(method = "tick",
+			  at = @At(
+					   value = "INVOKE",
+					   target = "Lnet/minecraft/block/entity/BrewingStandBlockEntity;craft(Lnet/minecraft/world/World;Lnet/minecraft/util/math/BlockPos;Lnet/minecraft/util/collection/DefaultedList;)V"
+			  )
+	)
+	private static void craftBrewingRecipe(World world, BlockPos pos, DefaultedList<ItemStack> slots) {
+		BrewingStandBlockEntity brewingStand = (BrewingStandBlockEntity) world.getBlockEntity(pos);
+		BrewingStandBlockEntityMixin recipeHolder = (BrewingStandBlockEntityMixin) (Object) brewingStand;
+		if (recipeHolder.quilt$recipe.matches(brewingStand, world)) {
+			recipeHolder.quilt$recipe.craft(brewingStand);
+
+			ItemStack ingredient = slots.get(3);
+
+			// TODO take into account recipe remainder api
+			ingredient.decrement(1);
+			if (ingredient.getItem().hasRecipeRemainder()) {
+				ItemStack remainder = new ItemStack(ingredient.getItem().getRecipeRemainder());
+				if (ingredient.isEmpty()) {
+					ingredient = remainder;
+				} else {
+					ItemScatterer.spawn(world, pos.getX(), pos.getY(), pos.getZ(), remainder);
+				}
+			}
+
+			slots.set(3, ingredient);
+			world.syncWorldEvent(1035, pos, 0);
+
+		} else {
+			craft(world, pos, slots);
+		}
+	}
+
+	@SuppressWarnings("ConstantConditions")
+	@Inject(method = "tick", at = @At(value = "FIELD", target = "Lnet/minecraft/block/entity/BrewingStandBlockEntity;fuel:I", opcode = Opcodes.PUTFIELD, ordinal = 1))
+	private static void modifyFuelUse(World world, BlockPos pos, BlockState state, BrewingStandBlockEntity brewingStand, CallbackInfo ci) {
+		var recipeHolder = (BrewingStandBlockEntityMixin) (Object) brewingStand;
+		if (recipeHolder.quilt$recipe.matches(brewingStand, world)) {
+			recipeHolder.fuel -= recipeHolder.quilt$recipe.getFuelUse() - 1; // Minus 1 because the base method already subtracts one
+		}
+	}
+
+	@SuppressWarnings("ConstantConditions")
+	@Inject(method = "tick",
+			at = @At(value = "FIELD",
+					 target = "Lnet/minecraft/block/entity/BrewingStandBlockEntity;brewTime:I",
+					 opcode = Opcodes.PUTFIELD,
+					 ordinal = 2,
+					 shift = At.Shift.AFTER
+			)
+	)
+	private static void modifyBrewTime(World world, BlockPos pos, BlockState state, BrewingStandBlockEntity brewingStand, CallbackInfo ci) {
+		var recipeHolder = (BrewingStandBlockEntityMixin) (Object) brewingStand;
+		if (recipeHolder.quilt$recipe.matches(brewingStand, world)) {
+			// A brew time of 1 would be the equivalent of a 0 tick recipe
+			recipeHolder.brewTime = Math.max(recipeHolder.quilt$recipe.getBrewTime(), 1);
+		}
+	}
+
+	@Inject(method = "isValid", at = @At(value = "RETURN", ordinal = 2), cancellable = true)
+	private void isValidPotionItem(int slot, ItemStack stack, CallbackInfoReturnable<Boolean> cir) {
+		cir.setReturnValue(stack.isIn(RecipeImpl.VALID_INPUTS) && this.getStack(slot).isEmpty());
+	}
+
+	@Inject(method = "readNbt", at = @At("TAIL"))
+	private void readBurnTimeAsInt(NbtCompound nbt, CallbackInfo info) {
```

Comment:
```suggestion
	private void readBrewTimeAsInt(NbtCompound nbt, CallbackInfo info) {
```

Reply:
You caught me! I copypasted from #163

Reply:
That's probably why my idiot brain went "hey isn't there a separate PR for this", yeah